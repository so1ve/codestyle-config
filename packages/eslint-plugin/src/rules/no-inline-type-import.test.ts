import { run } from "./_test";
import rule, { RULE_NAME } from "./no-inline-type-import";

const valid = ['import type { a } from "foo";'];
const invalid = [
	'import { type a } from "foo";',
	'import { type a, b } from "foo";',
	'import D, { type a, b } from "foo";',
];

run({
	name: RULE_NAME,
	rule,

	valid,
	invalid,
});
