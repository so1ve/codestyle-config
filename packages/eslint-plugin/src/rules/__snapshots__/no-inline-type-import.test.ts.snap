// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`no-inline-type-import > invalid > import { type a } from "foo"; 1`] = `"import type { a } from "foo";"`;

exports[`no-inline-type-import > invalid > import { type a, b } from "foo"; 1`] = `
"import type { a } from "foo";
import { b } from "foo";"
`;

exports[`no-inline-type-import > invalid > import D, { type a, b } from "foo"; 1`] = `
"import type { a } from "foo";
import D, { b } from "foo";"
`;
